pandas/core/indexes/interval.py

```diff
@@ -968,14 +968,21 @@ def delete(self, loc):
         return self._shallow_copy(new_left, new_right)
 
     def insert(self, loc, item):
-        if not isinstance(item, Interval):
-            raise ValueError('can only insert Interval objects into an '
-                             'IntervalIndex')
-        if not item.closed == self.closed:
-            raise ValueError('inserted item must be closed on the same side '
-                             'as the index')
-        new_left = self.left.insert(loc, item.left)
-        new_right = self.right.insert(loc, item.right)
+        if isinstance(item, Interval):
+            if not item.closed == self.closed:
+                raise ValueError('inserted item must be closed on the same '
+                                 'side as the index')
+            left_insert = item.left
+            right_insert = item.right
+        elif is_scalar(item) and isna(item):
+            # GH 18295
+            left_insert = right_insert = item
```

Comment:
yes u need to use a nan compat with left iow this could be a NaT

Reply:
i think we have a left.na_value iirc (might be spelled differently)

Reply:
or maybe the underlying already handles this in the insert

Reply:
Updated.  Procedure I'm following is "check if any type of NA value is passed -> raise if the wrong type of NA is passed".  I suppose I could just bypass this and only check if the right type of NA is passed, if that would be preferred.